### ISSUE TITLE
报错原因不清晰

### ISSUE THREAD

**Issue by rs-lsl** [PaddlePaddle/Paddle#36242]
在对Unet网络训练时，池化层会减小patchsize，但是在输入的patchsize较小时，过多的池化操作最后会使得patchsize变为1*1后仍进行池化，这时就会报错，但是并不是提示patchsize过小，而是提示OSERROR: (Externel) cudnn error， CUDNN_STATUS_BAD_PARAM, 很难找到bug所在，希望能改进一下
![)ECW_FH HCK3X( `QP9O`7](https://user-images.githubusercontent.com/66960705/135411183-45c73a25-7ae2-48c0-af56-90df6c89878d.png)

**Comment by rs-lsl** [comment]
![)ECW_FH HCK3X( `QP9O`7](https://user-images.githubusercontent.com/66960705/135411358-8bf15987-a14d-4dc2-a355-5e94350b515a.png)

**Comment by rs-lsl** [comment]
池化是通过步长为2的卷积实现的

**Comment by littletomatodonkey** [comment]
你好，图像大小和batch size是多少呢？感觉像是输入不合规导致的，或者方便提供一个可复现的单测吗？

**Comment by rs-lsl** [comment]
这是网络定义：
`        self.conv1 = nn.Conv2D(input_nc, d, 4, 2, 1)#input 3channels，output d channels,kernel size=4,stride=2,padding=1
        self.conv2 = nn.Conv2D(d, d * 2, 4, 2, 1)
        self.conv3 = nn.Conv2D(d * 2, d * 4, 4, 2, 1)
        self.conv4 = nn.Conv2D(d * 4, d * 8, 4, 2, 1)
        self.conv5 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)
        self.conv6 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)
        self.conv7 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)
        self.conv8 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)`
这是forward里的：
`        e1 = self.conv1(input)   # input: 64*64
        e2 = self.conv2(F.leaky_relu(e1, 0.2))
        e3 = self.conv3(F.leaky_relu(e2, 0.2))
        e4 = self.conv4(F.leaky_relu(e3, 0.2))
        e5 = self.conv5(F.leaky_relu(e4, 0.2))
        e6 = self.conv6(F.leaky_relu(e5, 0.2))
        e7 = self.conv7(F.leaky_relu(e6, 0.2))
        e8 = self.conv8(F.leaky_relu(e7, 0.2))`

如果输入大小为64*64， 那么在e7生成时就会报错，因为patchsize已经变成1*1了

**Comment by rs-lsl** [comment]
这是网络定义部分：

`       
        self.conv1 = nn.Conv2D(input_nc, d, 4, 2, 1)#input 3channels，output d channels,kernel size=4,stride=2,padding=1

        self.conv2 = nn.Conv2D(d, d * 2, 4, 2, 1)  

        self.conv3 = nn.Conv2D(d * 2, d * 4, 4, 2, 1)  

        self.conv4 = nn.Conv2D(d * 4, d * 8, 4, 2, 1)  

        self.conv5 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)

        self.conv6 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)

        self.conv7 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)

        self.conv8 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)`

这是forward函数里：

`        e1 = self.conv1(input)

        e2 = self.conv2(F.leaky_relu(e1, 0.2))

        e3 = self.conv3(F.leaky_relu(e2, 0.2))

        e4 = self.conv4(F.leaky_relu(e3, 0.2))

        e5 = self.conv5(F.leaky_relu(e4, 0.2))

        e6 = self.conv6(F.leaky_relu(e5, 0.2))   # 1*1

        e7 = self.conv7(F.leaky_relu(e6, 0.2))

        e8 = self.conv8(F.leaky_relu(e7, 0.2))`

如果输入大小为64，那么在e7生成时就会报错，因为 e6 的patchsize已经为 1 了。

**Comment by littletomatodonkey** [comment]
方便的话 可以传一个可以复现问题的python文件上来嘛？这样直接粘贴代码，格式是乱的~另外，我这边基于paddle2.1.3测试，似乎没有出现您说的问题

```
import paddle
import paddle.nn as nn
import paddle.nn.functional as F

class DemoNN(nn.Layer):
    def __init__(self, input_nc=3, d=8):
        super().__init__()
        self.conv1 = nn.Conv2D(input_nc, d, 4, 2, 1)
        self.conv2 = nn.Conv2D(d, d * 2, 4, 2, 1)  
        self.conv3 = nn.Conv2D(d * 2, d * 4, 4, 2, 1)  
        self.conv4 = nn.Conv2D(d * 4, d * 8, 4, 2, 1)  
        self.conv5 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)
        self.conv6 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)
        self.conv7 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)
        self.conv8 = nn.Conv2D(d * 8, d * 8, 4, 2, 1)

    def forward(self, x):
        e1 = self.conv1(x)
        e2 = self.conv2(F.leaky_relu(e1, 0.2))
        e3 = self.conv3(F.leaky_relu(e2, 0.2))
        e4 = self.conv4(F.leaky_relu(e3, 0.2))
        e5 = self.conv5(F.leaky_relu(e4, 0.2))
        e6 = self.conv6(F.leaky_relu(e5, 0.2))
        e7 = self.conv7(F.leaky_relu(e6, 0.2))
        e8 = self.conv8(F.leaky_relu(e7, 0.2))
        return e8

model = DemoNN(3, 8)
x = paddle.rand([1, 3, 4, 4])
x.stop_gradient = False
momentum = paddle.optimizer.Momentum(learning_rate=0.0003, parameters = model.parameters())
out = model(x)
out.backward()
momentum.step()
momentum.clear_grad()
```
